src/Symfony/Component/HttpClient/RetryableHttpClient.php

```diff
@@ -69,43 +69,64 @@ public function request(string $method, string $url, array $options = []): Respo
                 $context->setInfo('retry_count', $retryCount);
             }
 
-            if (null === $exception) {
-                if ($chunk->isFirst()) {
-                    $context->setInfo('retry_count', $retryCount);
+            if (null !== $exception) {
+                // check if the request has been sent
+                $uploaded = $context->getInfo('size_upload');
```

Comment:
I think this is going to be fragile.
We could instead consider that if the DNS resolution failed, then we can retry, but otherwise, we should consider only idempotent methods (configurable)
if ('' === info(primary_ip) => ok to retry unconditionnaly

Reply:
done